### DISCUSSION TITLE
cannot get file object in rest api

### DISCUSSION THREAD

**Discussion by dinfer** [directus/directus#12196] [Archive]
I'm using version 9.6.0 in a docker and creating a model like this:

<img width="934" alt="image" src="https://user-images.githubusercontent.com/11496403/158635710-abd65944-fdd1-4f0e-87b4-7908f2286f13.png">

Using rest api to query like this,`/items/product_categories?filter[banner][_neq]=null&fields[]=*,banner.*`, I cannot get the file object of `banner` field, described in the doc: https://docs.directus.io/reference/files/#the-file-object. 

<img width="709" alt="image" src="https://user-images.githubusercontent.com/11496403/158636934-a98479f2-6092-4180-948c-db22230b471a.png">

Is there a way to get the file object directly in the model query api, avoiding call files api again?

**Comment by rijkvanzanten** [comment]
Does the user you use have read access to Files?

**Reply by dinfer** [reply]
yes, I can query files directly using api, permissions are granted to public role.
<img width="946" alt="image" src="https://user-images.githubusercontent.com/11496403/158662643-12482dbd-c9f9-4fb8-a2bb-f2ffc7a49ed8.png">
<img width="794" alt="image" src="https://user-images.githubusercontent.com/11496403/158662775-b9379d0c-99e9-497f-8d82-054394b0c061.png">

**Reply by benhaynes** [reply]
I'm no dev, but is that `[]` supposed to be there in the `fields` parameter? Could you try this:

```
&fields=*,banner.*
```

Or try this as a sanity check:

```
&fields=*.*.*
```

You could also try adding an admin user's access token to bypass permissions (eg: `&access_token=your-token-here`)

**Reply by dinfer** [reply]
Thanks for your reply, I have double checked the permission, I think its OK.
I tried `&fields=*,banner.*` and `&fields=*.*.*`, it does not works. It seems a bug ?
<img width="863" alt="image" src="https://user-images.githubusercontent.com/11496403/158686464-bff951cf-8240-43a5-944d-d55febfd36c7.png">

**Reply by azrikahar** [reply]
That's odd, it does work on my end. Here you can see I show the example result when we have yet to grant read permission to `directus_files`, where `banner.*` has no effect as expected. But after I granted it, you can see the additional file object is retrieved with the exact same API call:

https://user-images.githubusercontent.com/42867097/158792732-99ea8643-4c1a-4839-a521-59d94606bc68.mp4

Hence I can only assume there's some form of setup issue here. Is the permission granted on the correct collection and Directus instance (in case you have multiple instances running)? Can you try checking with a fresh Directus instance?
